### PR TITLE
Prevent actions from being performed and icons from being shown when …

### PR DESCRIPTION
**Pull request by Revenaunt**
…there is no content.

If there is no content to a goal, the GoalTry won't be launched. The same way, if a goal has no actions, ChooseActions won't be launched. Icons in GoalTry are hidden if there is no content.

### CODE REVIEW

File: src/main/java/org/tndata/android/compass/activity/ChooseGoalsActivity.java [hunk 1]
```diff
@@ -321,11 +321,13 @@ public void goalSelected(Goal goal) {
...
-            intent.putExtra("goal", goal);
-            intent.putExtra("category", mCategory);
-            startActivity(intent);
+            if (goal.getBehaviorCount() > 0){
+                // Launch the GoalTryActivity (where users choose a behavior for the Goal)
+                Intent intent = new Intent(getApplicationContext(), GoalTryActivity.class);
+                intent.putExtra("goal", goal);
+                intent.putExtra("category", mCategory);
+                startActivity(intent);
+            }
```
Comment (bradmontgomery) [inline src/main/java/org/tndata/android/compass/activity/ChooseGoalsActivity.java]: Let's add an `else` block with a Toast that includes something like "Goal Added". We do that in other places, so let's be consistent.

File: src/main/java/org/tndata/android/compass/activity/LoginActivity.java [hunk 2]
```diff
@@ -116,6 +116,7 @@ private void handleBackStack() {
     }
 
     private void transitionToMain() {
+        Thread.dumpStack();
```
Comment (bradmontgomery) [inline src/main/java/org/tndata/android/compass/activity/LoginActivity.java]: & let's not forget about this.

File: src/main/java/org/tndata/android/compass/model/Goal.java [hunk 3]
```diff
@@ -15,6 +15,7 @@
     private String subtitle = "";
     private String outcome = "";
     private String icon_url = "";
+    private int behaviors_count;
```
Comment (bradmontgomery) [inline src/main/java/org/tndata/android/compass/model/Goal.java]: let's initialize `behviors_count` to 0.

**Comment by bradmontgomery** [comment]
@Revenaunt add the Toast, & initialized `Goal.behaviors_count` and I'm calling it :+1: . Once you make those 2 changes you can go ahead & merge.

**Comment by Revenaunt** [comment]
Goals with no behaviors get added without launching GoalTryActivity and behaviors with no actions get added without launching ChooseActionsActivity. Icons do not show up when there is no content to show.

**Pull request merged**